cmd/cody-gateway/internal/httpapi/completions/upstream.go

```diff
@@ -46,9 +46,10 @@ type upstreamHandlerMethods[ReqT UpstreamRequest] struct {
 	// validateRequest can be used to validate the HTTP request before it is sent upstream.
 	// Returning a non-nil error will stop further processing and return the given error
 	// code, or a 400.
+	// Second return value is a boolean indicating whether the request was flagged during validation.
 	//
 	// The provided logger already contains actor context.
-	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, error)
+	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, bool, error)
```

Comment:
Maybe this makes it a bit clearer as well? 
```suggestion
	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (httpStatus int, flagged bool, _ error)
```

Reply:
Implemented.